tableau/views.go

```diff
@@ -0,0 +1,72 @@
+package tableau
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// GetViewRequest encapsulates the request for getting a single View.
+type GetViewRequest struct {
+	ID      string
+	Filters []ViewFilter
+}
+
+type ViewFilter struct {
+	Name  string
+	Value string
+}
+
+// GetViewsRequest encapsulates the request for getting a single View.
+type GetViewsRequest struct {
+	PageSize   int
+	PageNumber int
+}
+
+type viewsService struct {
+	client *Client
+}
+
+func (dss *viewsService) GetViewData(ctx context.Context, getViewReq *GetViewRequest) (interface{}, error) {
+	requestFilter := ""
+	if len(getViewReq.Filters) > 0 {
```

Comment:
this will do but more idiomatic way to do this is Functional Options Pattern. You can use FOP for any kind of query parameters.